### ISSUE TITLE
Feature Request: Maintain a list of the queries and mutations being used 

### ISSUE THREAD

**Issue by EshaanAgg** [PalisadoesFoundation/talawa-admin#888]
**Is your feature request related to a problem? Please describe.**
Since the Talawa Admin client and the API is very tightly linked, it would make sense to maintain a list of all the graphql queries and mutations being used in the client currently in a single markdown file, so that the developers can easily refer to the same whenever some typeDefs are changed in the API, so as to be certain if the changes made there will make an breaking effect on the clients or not. 

**Describe the solution you'd like**
A markdown file listing all the queries and mutations being used. 

**Describe alternatives you've considered**
NA

**Approach to be followed (optional)**
NA

**Additional context**
NA

**Comment by EshaanAgg** [comment]
@palisadoes I would like to work on this issue.

**Comment by palisadoes** [comment]
1. Is there a way to automate this? It sounds like you are going to do a manual approach which will always be out of date.
1. Would a better approach be to refactor the code and its comments so that the queries and mutations are more easily found in the automated documentation?

**Comment by EshaanAgg** [comment]
1. I tried to find automated packages but wasn't able to find any. I think so manual approach would be the best for the same, and we can configure an additional check in the workflow if any of the files hosting the GraphQL queries and mutations are changed in a PR, then the user would get an automated message to update the documentation in the respective file when they open a PR.

2. I don't think there is a need for the same at the moment. Once we have this list ready in both the clients, we can then update the API with comments to show that this query/mutation is being used in this client currently, and then the same would automatically become a part of the automated documentation!

**Comment by palisadoes** [comment]
Wouldn't it be better to document which apps use the mutations and queries in the API code itself? That way the documentation would be automatically generated by TSDoc. This seems like unnecessary work for this repo. We need to minimize the manual update of documentation across repos.

**Comment by EshaanAgg** [comment]
The problem with that is practically no contributor would like to hop over to another repo and update the documentation whenever they want to introduce a new query/mutation to a client. The documentation should reside where the code is.

I also completely agree with the problem that you have described of manual updation of stuff, and the simplest solution to automate the same would be to use a custom linter or a script. But the same would be highly exploratory in nature, and would require comparitvely a lot of time to be devoted to the same.

**Comment by palisadoes** [comment]
Yes, but if we document these in the API code in a TSDocs compatible way, then the docs.talawa.io site to get the information required.

- https://docs.talawa.io/docs/category/mutations 

I'd think that the time would be better spent updating the descriptions in the API code. Here is a good example

- https://docs.talawa.io/docs/schema/mutations/reject-admin

We could then add Google Search to docs.talawa.io to make things easy to find.

- https://support.google.com/programmable-search/answer/12423533?hl=en

Is this a viable alternative in your opinion?

**Comment by EshaanAgg** [comment]
Got it @palisadoes. I feel that this can work! Will open a relevant issue in API.
